pages/food-groups/index.tsx

```diff
@@ -0,0 +1,280 @@
+  const [selectedDraggableType, setSelectedDraggableType] = useState(
+    GROUPS.NONE
+  )
+
+  // GAME STATE
+  const [hoverType, setHoverType] = useState(GROUPS.DEFAULT)
+  const [roundCounter, setRoundCounter] = useState(0)
+  const [correctDraggables, setCorrectDraggables] = useState(() =>
+    newArray(false)
+  )
+  const [wheelEnabled, setWheelEnabled] = useState(true)
+
+  // This keeps the number of wins while signed out so when the user logs in it gets added to their score
+  const [roundCounterSignedOut, setRoundCounterSignedOut] = useState(0)
+
+  // Draggable Logic
+  const [overridePosition, setOverridePosition] = useState(ORIGIN_VECTOR2)
+
+  const [currentCharSet, setCharSet] = useState<
+    FoodGroupCharacterImageDynamic[]
+  >(() => generateCharacterSet(notion_character_data))
+  const [nextCharSet, setNextCharSet] = useState<
+    FoodGroupCharacterImageDynamic[]
+  >(() => generateCharacterSet(notion_character_data))
+
+  const draggablePositions_1: State_<Vector2>[] = []
+  const draggablePositions_2: State_<Vector2>[] = []
+  for (let i = 0; i < 5; i++) {
+    draggablePositions_1[i] = useState<Vector2>(() => ORIGIN_VECTOR2)
+    draggablePositions_2[i] = useState<Vector2>(() => ORIGIN_VECTOR2)
+  }
+
+  const [draggingStates, setDraggingStates] = useState<DRAGGING_STATE[]>(() =>
+    newArray(DRAGGING_STATE.START)
+  )
+
+  const [switchSetFlag, setswitchSetFlag] = useState(true)
+
+  const draggableZoneRef = useRef<any>(undefined)
+  const draggableZoneWheelRef = useRef<any>(undefined)
+  const [draggableZone, setDraggableZone] = useState<DOMRect | undefined>()
+
+  useEffect(() => {
+    // Checking if the draggable parent element's rect got initialised
+    if (!draggableZoneRef.current)
+      throw new Error(
+        'reference to parent div containing foodgroup and draggables didnt return .current'
+      )
+    if (!draggableZoneWheelRef.current)
+      throw new Error('Wheel Ref did not return current')
+    setDraggableZone(draggableZoneRef.current.getBoundingClientRect())
+
+    window.addEventListener('resize', () => {
+      if (!draggableZoneRef.current)
+        throw new Error(
+          'reference to parent div containing foodgroup and draggables didnt return .current'
+        )
+      setDraggableZone(draggableZoneRef.current.getBoundingClientRect())
+    })
+
+    window.addEventListener('resize', () => {
+      if (!draggableZoneWheelRef.current)
+        throw new Error('draggableZoneWheelRef not initialised?')
+    })
+  }, [])
+
+  useEffect(() => {
+    if (user && roundCounterSignedOut > 0) {
+      updateAchievementsDocument?.({
+        [ACHIEVEMENT.DRAG_DROP_WIN_COUNT]:
+          achievements[ACHIEVEMENT.DRAG_DROP_WIN_COUNT] + roundCounterSignedOut,
+        // For now, achivements is just the amount of game wins
+        [ACHIEVEMENT.ACHIEVEMENT_COUNT]:
+          achievements[ACHIEVEMENT.ACHIEVEMENT_COUNT] + roundCounterSignedOut
+      })
```

Comment:
There are some major changes made to `FirebaseContext`. Please update the codes that are using the values and functions from this context.